### PR TITLE
Bump knative/pkg to 1.11

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Bump knative/pkg to the latest, 1.11

This also updates the min kubernetes version to 1.25.

/kind misc

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Bump knative.dev/pkg to 1.11 so the Kubernetes min version is now 1.25
```

### CODE REVIEW

**Comment by vdemeester** [comment]
Ah this needs a change in plumbing..

**Comment by vdemeester** [comment]
This will need https://github.com/tektoncd/plumbing/pull/1520
cc @afrittoli @tektoncd/plumbing-maintainers

File: pkg/reconciler/events/cloudevent/cloudeventclient.go [hunk 1]
```diff
@@ -33,7 +33,6 @@ func init() {
 	injection.Default.RegisterClient(func(ctx context.Context, _ *rest.Config) context.Context {
 		return withCloudEventClient(ctx)
 	})
-	injection.Dynamic.RegisterDynamicClient(withCloudEventClient)
```
Comment (afrittoli) [inline pkg/reconciler/events/cloudevent/cloudeventclient.go:36]: Do you know what replaced this?

Comment (vdemeester) [reply pkg/reconciler/events/cloudevent/cloudeventclient.go:36]: Nothing, it was removed altogether. See https://github.com/knative/pkg/issues/2739#issuecomment-1550452957. It was not really used anywhere.

Comment (lbernick) [reply pkg/reconciler/events/cloudevent/cloudeventclient.go:36]: would you mind adding a note about this to the commit message/pr description?

**Review by lbernick — APPROVED** [review]

File: pkg/reconciler/events/cloudevent/cloudeventclient.go (hunk 1 above)
Comment (vdemeester) [reply pkg/reconciler/events/cloudevent/cloudeventclient.go:36]: Actually, I am already linking it from the commit message (and saying it's not used), is that good enough ?

Comment (lbernick) [reply pkg/reconciler/events/cloudevent/cloudeventclient.go:36]: Ah my bad! I normally just look at the PR description and assume the commit message is the same :) Can you squash your commits?

Comment (vdemeester) [reply pkg/reconciler/events/cloudevent/cloudeventclient.go:36]: @lbernick I can 🙃 I wanted to split it into several part though, one for the bump, one the updating the kubernetes min version and the rest for fixing some changes, but.. I can also make it one big commit 😛

**Review by afrittoli — COMMENTED** [review]
Thank you @vdemeester !
/lgtm

**Comment by vdemeester** [comment]
arg.. needs a rebase 😅

**Pull request merged**